### PR TITLE
refactor: upgrade to latest version of entur sdk for mobile token

### PR DESCRIPTION
**Pull request by hanne-at-atb**
Contributes to closing: https://github.com/AtB-AS/kundevendt/issues/3741

See related changes to token-server: https://github.com/AtB-AS/abt-token-server/pull/44

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -43,12 +43,16 @@ target 'app' do
...
   if ENV['GH_ACCESS_TOKEN']
-    pod 'AbtTokenCoreSDK', :git => 'https://' + ENV['GH_ACCESS_TOKEN'] + '@github.com/AtB-AS/temp-token-core-ios-lib.git'
+  # TODO: Point to release branch
+    pod 'TokenCore', :git => 'https://' + ENV['GH_ACCESS_TOKEN'] + '@github.com/entur/abt-mobile-client-sdk'
+    pod 'TokenStateReactNativeLib', :git => 'https://' + ENV['GH_ACCESS_TOKEN'] + '@github.com/entur/abt-mobile-client-sdk'
+    pod 'AbtProtobufTokenIosStubs', :git => 'https://' + ENV['GH_ACCESS_TOKEN'] + '@github.com/entur/abt-protobuf-pods.git', :tag => '2.661.0'
   else
     # If no GH_ACCESS_TOKEN in env, then probably locally where the user uses personal ssh key
-    pod 'AbtTokenCoreSDK', :git => 'https://github.com/AtB-AS/temp-token-core-ios-lib.git'
+    pod 'TokenCore', :git => 'git@github.com:entur/abt-mobile-client-sdk.git', :branch => 'develop'
```
Comment (jorelosorio) [inline ios/Podfile:52]: This pod file is also used for production or staging and the branch is pointing to develop, is that correct or should point to master instead?

Comment (gorandalum) [reply ios/Podfile:52]: The `else` clause here is when we are on local development, and then we use the `develop` branch. When not local development, it should use the `master` branch. But your comment made me think of something, we probably need to pin the version in the staging/store build to a specific release tag. I'll look into it!

Comment (gorandalum) [reply ios/Podfile]: Updated this now:
https://github.com/AtB-AS/mittatb-app/pull/3448/commits/983353e521ada433521869ea25a5a7fe0c23f5aa

**Review by jorelosorio — APPROVED** [review]
I went through the code and I find it well, good work :)

**Review by rosvik — APPROVED** [review]

**Pull request merged**